daily_milestone.go

```diff
@@ -79,6 +79,25 @@ func LoggerSetup(info io.Writer) {
 	logger = log.New(info, "INFO: ", log.Ldate|log.Ltime|log.Lshortfile)
 }
 
+// Function to get last day of the month
+func lastDayMonth(year int, month int, timezone *time.Location) time.Time {
+	t := time.Date(year, time.Month(month)+1, 0, 0, 0, 0, 0, time.UTC)
+	return t
+}
+func lastDayWeek(lastDay time.Time) (int, int, time.Time) {
```

Comment:
So when it's not Monday it will iterate back until it reaches Monday.
Tuesday -> -1 -> Monday.
Wouldn't that give back the first day of the week?

(the else statement)
When it's already Monday, then you add a whole week. wouldn't that miss the first weekly milestone?
2017-12-18 is a Monday. It would go into the else loop and jump to Monday 2017-12-25.

Reply:
Yes that's what I should look into.